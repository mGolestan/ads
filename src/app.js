// @flow
import express from "express";
import bodyParser from "body-parser";
import compress from "compression";
import routes from "./routes";
import helmet from "helmet";
import cors from "cors";
import loadEnv from "./utils/loadEnv";
import { notFound, errResponse, logRequest } from "./middlewares";

const app = express();

app
  .use(
    helmet({
      dnsPrefetchControl: false,
      frameguard: false,
      noCache: true
    })
  )
  .use(compress())
  .use(cors("*"))
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .use(logRequest)
  .use("/", routes)
  .use(notFound)
  .use(errResponse);

module.exports = app;
