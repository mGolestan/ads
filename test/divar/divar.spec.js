// @flow

import request from "supertest";

import app from "../../src/app";

describe("GET /scrape/divar", () => {
  it(
    "returns a valid result on a valid request",
    () => {
      const city = "tehran";
      const q = "میز";
      return request(app)
        .get(`/scrape/divar?city=${city}&q=${q}`)
        .expect(200);
    },
    10000
  );

  it("returns BadRequest on an invalid city", () => {
    const city = "";
    const q = "میز";
    return request(app)
      .get(`/scrape/divar?city=${city}&q=${q}`)
      .expect(400);
  });

  it("returns BadRequest on an invalid q parameter", () => {
    const city = "terhan";
    const q = "";
    return request(app)
      .get(`/scrape/divar?city=${city}&q=${q}`)
      .expect(400);
  });

  it("returns BadRequest if not entered url querys", () => {
    return request(app)
      .get(`/scrape/divar`)
      .expect(400);
  });
});
