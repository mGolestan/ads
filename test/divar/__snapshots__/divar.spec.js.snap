// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /scrape/divar returns a valid result on a valid request 1`] = `
Object {
  "results": Array [
    Object {
      "contact": "https://api.divar.ir/v1/posts/HmfgJ8cul/contact",
      "description": "صفر کارتکس تمامی رنگ ها موجود است جهت اطلاع بیشتر تماس بگیرید با تشکر",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531396835/HmfgJ8cul.jpg",
      "price": 53500000,
      "site": "Divar",
      "timeStamp": 1531396835,
      "title": "تندر 90 e2",
      "token": "HmfgJ8cul",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/jW_PHaeQu/contact",
      "description": "مگان۱۶۰۰E2 فول دارای۳ مانیتور دوربین دنده عقب شیشه دودی دارای رینگ اسپرت تعویض تسمه تایم بسیار تمیز فوری فروشی",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531395947/jW_PHaeQu.jpg",
      "price": 85000000,
      "site": "Divar",
      "timeStamp": 1531395947,
      "title": "مگان ۱۶۰۰  E2 فول",
      "token": "jW_PHaeQu",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/rsVk-tEhX/contact",
      "description": "با سلام خدمت شما دوستان گرامی، فنی و سواری و اتاق سالم در حد صفر می باشد، فقط قوس گلگیر های عقب رنگ شدگی جزئی دارد بدون ضربه و تصادف، ۶ ماه بیمه دارد از ابتدا تخفیف، یکسال معاینه فنی، ۴ حلقه لاستیک نو...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531395238/rsVk-tEhX.jpg",
      "price": -1,
      "site": "Divar",
      "timeStamp": 1531395238,
      "title": "ال نود فول e2 پارس خودرو استثنایی ۸۹",
      "token": "rsVk-tEhX",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/17QYnsju2/contact",
      "description": "فنی کاملا سالم عقب جلو سالم دو لکه رنگ جزعی 6 ماه بیمه",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531395172/17QYnsju2.jpg",
      "price": 23000000,
      "site": "Divar",
      "timeStamp": 1531395172,
      "title": "تندر 90 e2 نیمه فول 86",
      "token": "17QYnsju2",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/dRgDqAUjG/contact",
      "description": "فوق العاده سالم و کم کارکرد؛ رنگ شدگی فقط مقداری از درب جلو ولبه گلگیرعقب دارد ؛ E2 فول پارس خودرو ؛ دو ایربگ؛ سند دست اول ؛ بیمه تا برج 9 تخفیف بیمه 6 سال ؛ روکش صندلی چرمی نو انداختم ؛ لاستیک ها 80 ...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531390547/dRgDqAUjG.jpg",
      "price": 33800000,
      "site": "Divar",
      "timeStamp": 1531390547,
      "title": "ال ۹۰ مدل ۹۱ E2 فول پارس خودرو",
      "token": "dRgDqAUjG",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/VWM58SXv5/contact",
      "description": "مدل 97 سفید گیربکس جدید jhq پلاک شده ولی پلاک پرچ نشده تحویلی برج 4",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531387108/VWM58SXv5.jpg",
      "price": 56800000,
      "site": "Divar",
      "timeStamp": 1531387108,
      "title": "تندر 90 E2 فول پارس خودرویی jhq",
      "token": "VWM58SXv5",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/E1XpAsJkk/contact",
      "description": "بسبارتمیز وسالم پارس خودرویی E2 تک برگ سند فوق العاده بکر وعالی",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531386454/E1XpAsJkk.jpg",
      "price": 35200000,
      "site": "Divar",
      "timeStamp": 1531386454,
      "title": "النودE2پارس خودرویی",
      "token": "E1XpAsJkk",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/6WLfqOlOm/contact",
      "description": "ال نود e2فول پارس خودرو بدون رنگ و بدون خوردگی،بدون تعویض بشرط کارشناسی از شما بیمه تا آخر برج10 و تخفیف بیمه کامل میباشد کارکرد ماشین واقعی و بشرط دیاگ تمام لوازم های مصرفی ماشین مثل روغن فیلتر هوا ی...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531383820/6WLfqOlOm.jpg",
      "price": 32500000,
      "site": "Divar",
      "timeStamp": 1531383820,
      "title": "ال نود e2فول پارس بدون رنگگ",
      "token": "6WLfqOlOm",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/a1Va-m5f9/contact",
      "description": "سفید، گیربکس جدید",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531379217/a1Va-m5f9.jpg",
      "price": 54000000,
      "site": "Divar",
      "timeStamp": 1531379217,
      "title": "خودرو ال ٩٠ e2  پارس خودرو",
      "token": "a1Va-m5f9",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/SJ2lzkB1m/contact",
      "description": "کاملا بی رنگ ، سنسور دنده عقب ، آینه با دوربین عقب ، تو دوزی کامل چرم نو نو مشکی ضبط پایونیر . سالم ترین مگان سال ٩٠ فنی در حد صفر e2 بیمه بدنه ١ سال کامل بیمه ثالث ٦ ماه",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531359317/SJ2lzkB1m.jpg",
      "price": 69000000,
      "site": "Divar",
      "timeStamp": 1531359317,
      "title": "e2 مگان ٩٠ دنده ای",
      "token": "SJ2lzkB1m",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/k9lSI90JE/contact",
      "description": "فنی سالم .رخ عالی .شیشه دودی .دزدگیر معمولی .پاور ویندو .ظاهر اسپرت .بیمه بدنه با تخفیف کامل. بیمه ثالث دولکه رنگ به علت زیبایی",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531349655/k9lSI90JE.jpg",
      "price": 30000000,
      "site": "Divar",
      "timeStamp": 1531349655,
      "title": "تندر 90 E2 مشکی",
      "token": "k9lSI90JE",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/Lv9Z519vW/contact",
      "description": "خودرو بدون رنگ وتصادف بوده رنگ خودرونوک مدادی متالیک ۴حلقه لاستیک ۲۰۵هشتاد درصد بیمه۴ماه و از اولش تخفیف دارد ضبط CDو فلش خور دارد",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531349072/Lv9Z519vW.jpg",
      "price": 30000000,
      "site": "Divar",
      "timeStamp": 1531349072,
      "title": "تندر پارس خودرویی E2فول",
      "token": "Lv9Z519vW",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/TN_CZe7Dc/contact",
      "description": "سلام . توضیحات کامل است . لطفا ابتدا ملاحظه کنید سپس در صورت تمایل تماس بگیرید . متشکرم . به دلیل مشغله کاری از پاسخگویی به SMS معذورم . عزیزان دلال ٬ واسطه و کاسب تماس نگیرند . تندر ۹۰ سفید E2 فول مد...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531363540/TN_CZe7Dc.jpg",
      "price": 35400000,
      "site": "Divar",
      "timeStamp": 1531363540,
      "title": "تندر۹۰سفیدE2فول مدل۹۱بدون خط وخش",
      "token": "TN_CZe7Dc",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/GjN4ndPrk/contact",
      "description": "سفید گیربکس جدید JHQ تحویل ۹۷/۴/۱۳ همه مدارک آماده برای تعویض پلاک ایران خودرو فقط مصرف کننده باتشکر",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531392055/GjN4ndPrk.jpg",
      "price": 54500000,
      "site": "Divar",
      "timeStamp": 1531392055,
      "title": "تندر ۹۰ e2 صفر خشک",
      "token": "GjN4ndPrk",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/D2sMwhy2u/contact",
      "description": "ال ۹۰ e2 پارس خودرویی سفید گیربکس جدیدصفر پلاک شده اماده محضر محدوده تهرانسر",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531348129/D2sMwhy2u.jpg",
      "price": 56600000,
      "site": "Divar",
      "timeStamp": 1531348129,
      "title": "ال ۹۰ e2 صفر  پارس خودرویی گیربکس جدید",
      "token": "D2sMwhy2u",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/uxfszouBQ/contact",
      "description": "یک گلگیر جلو-جفت دور گلگیر عقب به شرط بالای گلگیر نباشه به شرطی که مالیدگی بوده باشه بدون ضعف عقب جلو خط و خش کمی دارد بیمه معاینه کامل لاستیک٧٠./. تک برگ سند ماشین خونگی سند آماده فنی مشابه صفر",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531346561/uxfszouBQ.jpg",
      "price": 32500000,
      "site": "Divar",
      "timeStamp": 1531346561,
      "title": "ال٩٠E2 مشکی مدل٩١",
      "token": "uxfszouBQ",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/fCh0qBwv1/contact",
      "description": "ال نود مدل ۸۷ پارس خودرویی بیمه ۱۱ ماه معاینه فنی تا شهریور ماه گیربکس اروپایی شتاب بسیار عالی جلو بندی کاملا نو یه لکه رنگ جلو عقب پلمپ رو کش صندلی نو شیشه ها دودی لاستیک ها ۸۰ درصد ۲۰۵ کلیه لوازم مص...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531346234/fCh0qBwv1.jpg",
      "price": 32000000,
      "site": "Divar",
      "timeStamp": 1531346234,
      "title": "ال ۹۰ E 2 فول",
      "token": "fCh0qBwv1",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/BRj42oQeA/contact",
      "description": "با سلام ال نود سفید بسیار تمیز بدون خط و خش دو حلقه لاستیک جلو نو لاستیک عقب ٨٠ درصد بیمه یک سال تخفیف کامل،دزدگیر تصویرى،سنسور دنده عقب،روکش صندلى نو دو لکه جزئى رنگ روى برامدگى گلگیر عقب بدون تصادف ...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531343758/BRj42oQeA.jpg",
      "price": 36500000,
      "site": "Divar",
      "timeStamp": 1531343758,
      "title": "l 90 e2 full",
      "token": "BRj42oQeA",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/YOacOxcEw/contact",
      "description": "ال٩٠E2فول سفید شیشه دودى کفپوش کامل دوزدگیر قفل ضد سرقت طوریى زیر شیشه بدون رنکه بدون خط خش تعویض روغن انجام شد بدون ١٠٠٠ خرجه",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531343691/YOacOxcEw.jpg",
      "price": 52900000,
      "site": "Divar",
      "timeStamp": 1531343691,
      "title": "ال٩٠فول E2",
      "token": "YOacOxcEw",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/FH7XuNUY0/contact",
      "description": "سلام ماشین بیرنگه و بدون تعویض بدون ضربه و بی خط خش شش ماه بیمه دارد و تخفیف بیمش کامله ۸۰ درصد لاستیک دارد جلوبندی تازه سرویس شده فنیش سالمه بی سروصداس لوازم مصرفی ازجمله تسمه تایم دیسک صفحه لنت جلو ...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531342584/FH7XuNUY0.jpg",
      "price": 23800000,
      "site": "Divar",
      "timeStamp": 1531342584,
      "title": "تندر ۹۰ مدل ۱۳۸۷ e2 بیرنگ",
      "token": "FH7XuNUY0",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/kIIAT8_IM/contact",
      "description": "فنی سالم به شرط بی سرو صدا بی رنگ",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531338196/kIIAT8_IM.jpg",
      "price": -1,
      "site": "Divar",
      "timeStamp": 1531338196,
      "title": "ال90 مدل 90 E2",
      "token": "kIIAT8_IM",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/R_aPiwdFe/contact",
      "description": "تندر 90 e2 پارس خودرو صفر گیربکس جدید کلیه مدارک اماده خودرو اماده تحویل در پارکینگ منزل می باشد.",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531335804/R_aPiwdFe.jpg",
      "price": 57200000,
      "site": "Divar",
      "timeStamp": 1531335804,
      "title": "تندر90 e2 پارس خودرو",
      "token": "R_aPiwdFe",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/IqXz70JL2/contact",
      "description": "✔ ال۹۰ نقره ای مدل پارس خودرویی (یک و نیم میلیون گران تر از مدل ایران خودرویی) . ✔ مدل E2فول(آیینه ها برقی ، ۲ تا ایربگ ،شیشه های عقب و جلو برقی و دکمه قفل کن شیشه ها) ✔به هیچ وجه تصادف نداشته ماشین ک...",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531333923/IqXz70JL2.jpg",
      "price": 31500000,
      "site": "Divar",
      "timeStamp": 1531333923,
      "title": "ال۹۰ E2فول بی رنگ آخر ۸۷پارس خودرویی",
      "token": "IqXz70JL2",
    },
    Object {
      "contact": "https://api.divar.ir/v1/posts/_o_SFr4xJ/contact",
      "description": "تندر ۹۰ پارس خودرو بدون تصادف بدون رنگ بیمه از ۹۳ تخفیف کامل بیمه تا ۱۱.۵.۹۷ ۴حلقه لاستیک ۱۰۰ درصد تسمه تایم ۷۵۰۰۰ تعویض",
      "image": "https://s100.divarcdn.com/static/thumbnails/1531327808/_o_SFr4xJ.jpg",
      "price": 47000000,
      "site": "Divar",
      "timeStamp": 1531327808,
      "title": "تندر ۹۰ e2 پارس خودرو",
      "token": "_o_SFr4xJ",
    },
  ],
  "version": 1,
}
`;
